pyam/plotting.py

```diff
@@ -375,19 +375,103 @@ def stack_plot(df, x='year', y='value', stack='variable',
     # long form to one column per bar group
     _df = reshape_bar_plot(df, x, y, stack)
 
+    # Line below is for interpolation. On datetimes
+    # I think you'd downcast to seconds first and
+    # then cast back to datetime at the end..?
+    _df.index = _df.index.astype(float)
+
+    time_original = _df.index.values
+    first_zero_times = pd.DataFrame(index=["first_zero_time"])
+    both_positive_and_negative = _df.apply(
+        lambda x: (x >= 0).any() and (x < 0).any()
+    )
+
+    for col in _df.loc[:, both_positive_and_negative]:
```

Comment:
I get what this is doing, but it still may be worth adding a small comment at the top explaining it.

I think this could be a little tighter using `np.argmax` or, for example

```
positive = values[values >= 0]
negative = values[values <= 0]
crosses = positive[::-1] & negative[1::]
```

and then you can calculate the time for each index where crosses is true